enterprise/coderd/schedule/template.go

```diff
@@ -195,6 +195,24 @@ func (s *EnterpriseTemplateScheduleStore) Set(ctx context.Context, db database.S
 			return xerrors.Errorf("get updated template schedule: %w", err)
 		}
 
+		// Update all workspace's TTL using this template if either of the following:
+		//   - The template's AllowUserAutostop has just been disabled
+		//   - The template's TTL has been modified and AllowUserAutostop is disabled
+		if (!opts.UserAutostopEnabled && opts.UserAutostopEnabled != tpl.AllowUserAutostop) ||
```

Comment:
```suggestion
		if (!opts.UserAutostopEnabled && tpl.AllowUserAutostop) ||
```
This can be simplified since we're already checking the boolean condition.

Also, shouldn't we use `template` instead of `tpl` here for consistency? I'm confused why we use `template.AllowUserAutostop` on the line below but `tpl` here.

Reply:
Thanks for the simplification!

The reason for using `tpl` here is that we want to compare against the previous value, but `template` has the current value. I've decided to drop using `template` in favour of using `opts` instead to help reduce confusion though!

That's helped simplify (at least to me) the query even further, going from

```go
if (!opts.UserAutostopEnabled && A) || (!opts.UserAutostopEnabled && B)
// to
if !opts.UserAutostopEnabled && (A || B)
```